File: src/agent_memory/enums.rs
use serde::{Deserialize, Serialize};

/// High-level governed memory types.
#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum MemoryType {
    Trace,
    Episode,
    Fact,
    Preference,
    GoalState,
}

/// Current status of an explicit belief.
#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum BeliefStatus {
    Active,
    Disputed,
    Stale,
    Retracted,
}

/// Intent classes for governed retrieval.
#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum QueryIntent {
    CurrentFact,
    HistoricalFact,
    PreferenceLookup,
    TaskState,
    EpisodicRecall,
    SemanticBackground,
}

/// Trust source of a memory observation.
#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum SourceType {
    Chat,
    File,
    Tool,
    System,
    External,
}

/// Visibility / applicability scope.
#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum Scope {
    Private,
    Task,
    Project,
    Shared,
}

/// Result of the promotion gate.
#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum PromotionDecision {
    Reject,
    StoreTrace,
    Promote,
}

/// Belief state transition kinds.
#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum BeliefAction {
    Reinforce,
    Update,
    Dispute,
    Retract,
}

File: agent_memory/enums.py
from __future__ import annotations

from enum import Enum


class MemoryType(str, Enum):
    TRACE = 'trace'
    EPISODE = 'episode'
    FACT = 'fact'
    PREFERENCE = 'preference'
    GOAL_STATE = 'goal_state'


class BeliefStatus(str, Enum):
    ACTIVE = 'active'
    DISPUTED = 'disputed'
    STALE = 'stale'
    RETRACTED = 'retracted'


class QueryIntent(str, Enum):
    CURRENT_FACT = 'current_fact'
    HISTORICAL_FACT = 'historical_fact'
    PREFERENCE_LOOKUP = 'preference_lookup'
    TASK_STATE = 'task_state'
    EPISODIC_RECALL = 'episodic_recall'
    SEMANTIC_BACKGROUND = 'semantic_background'


class SourceType(str, Enum):
    CHAT = 'chat'
    FILE = 'file'
    TOOL = 'tool'
    SYSTEM = 'system'
    EXTERNAL = 'external'


class Scope(str, Enum):
    PRIVATE = 'private'
    TASK = 'task'
    PROJECT = 'project'
    SHARED = 'shared'


class PromotionDecision(str, Enum):
    REJECT = 'reject'
    STORE_TRACE = 'store_trace'
    PROMOTE = 'promote'


class BeliefAction(str, Enum):
    REINFORCE = 'reinforce'
    UPDATE = 'update'
    DISPUTE = 'dispute'
    RETRACT = 'retract'
